compiler/rustc_query_system/src/dep_graph/mod.rs

```diff
@@ -81,23 +75,39 @@ pub trait DepContext: Copy {
     }
 
     /// Load data from the on-disk cache.
-    fn try_load_from_on_disk_cache(self, dep_node: DepNode<Self::DepKind>) {
+    fn try_load_from_on_disk_cache(self, dep_node: DepNode) {
         let cb = self.dep_kind_info(dep_node.kind);
         if let Some(f) = cb.try_load_from_on_disk_cache {
             f(self, dep_node)
         }
     }
 }
 
+pub trait Deps {
+    /// Execute the operation with provided dependencies.
+    fn with_deps<OP, R>(deps: TaskDepsRef<'_>, op: OP) -> R
+    where
+        OP: FnOnce() -> R;
+
+    /// Access dependencies from current implicit context.
+    fn read_deps<OP>(op: OP)
+    where
+        OP: for<'a> FnOnce(TaskDepsRef<'a>);
+
+    const DEP_KIND_NULL: DepKind;
+    const DEP_KIND_RED: DepKind;
```

Comment:
Could we make those associated constants on `DepKind` with values 0 and 1, and have `rustc_middle` check the consistency?

Reply:
I think it makes sense to delay dealing with these until multiple crates can define dep kinds.